PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/components/ServerExplorerToolWindowFactory.java

```diff
@@ -223,26 +224,31 @@ private Node getTreeNodeOnMouseClick(JTree tree, TreePath treePath) {
     }
 
     private JPopupMenu createPopupMenuForNode(Node node) {
-        JPopupMenu menu = new JPopupMenu();
-
-        for (final NodeAction nodeAction : node.getNodeActions()) {
-            if (!nodeAction.isEnabled()) {
-                continue;
-            }
-            JMenuItem menuItem = new JMenuItem(nodeAction.getName());
-            menuItem.setEnabled(nodeAction.isEnabled());
-            if (nodeAction.getIconPath() != null) {
-                menuItem.setIcon(UIHelperImpl.loadIcon(nodeAction.getIconPath()));
+        final JPopupMenu menu = new JPopupMenu();
+        final Map<Integer, List<NodeAction>> actionsMap = node.getNodeActions().stream().collect(Collectors.groupingBy(NodeAction::getGroup));
```

Comment:
As far as I know, `groupingBy` is order-stable. and LinkedHashMap is ordered.
you could do all the sorting/grouping here to make code simpler and more readable:
refer: 
  https://howtodoinjava.com/java8/sort-stream-multiple-fields/  
  https://stackoverflow.com/a/49283318

Reply:
Yes, changed to get sortedActionsGroupMap first and then convert to menu items in latest pr